### ISSUE TITLE
<C-y> is mapping to <C-z>

### ISSUE THREAD

**Issue by mqudsi** [martanne/vis#623]
I've run into an issue where a physical `<C-y>` keypress is being interpreted as `<C-z>` instead. I've tried to manually remap `<C-y>` via `:map normal <C-y> vis-window-slide-up` but that has not helped; each time `<C-y>` is pressed, the editor is sent to the background.

This happens regardless of the terminal emulator being used or the `TERM` environment variable setting.

**Comment by martanne** [comment]
Are you sure this not due to some external component e.g. your windowing system?

Vis relies on the keys as reported by [libtermkey](http://www.leonerd.org.uk/code/libtermkey/). If you build libtermkey from source you will find a `demo` application which prints the keys as you enter them. Does it work as expected?

**Comment by mqudsi** [comment]
Yeah, the demo in libtermkey runs just fine, which makes sense since other libtermkey applications (e.g. neovim) don't have this issue and interpret `<C-y>` just fine when run in the same session.

```
mqudsi@QBook ~/R/libtermkey> ./demo
Termkey in RAW mode
Key <C-y>
Key <C-c>
```

I can't get this behavior to reproduce on my WSL machine, only on macOS. I thought it might have something to do with the BSD underpinnings on OS X, but vis fails to run on FreeBSD 11 (hangs on start).

**Comment by martanne** [comment]
This sounds rather strange, vis' low level input handling should in principle be similar to the mentioned demo application. The only major difference being that libtermkey and nucurses might fight over the underlying terminal state.

Any chance you can debug this yourself? I don't have access to a macOS system...

As for the BSDs, vis seems to work fine on OpenBSD, I'm a bit surprised that isn't the case for FreeBSD. Patches to fix that would certainly be welcome too.

**Comment by mqudsi** [comment]
I'll poke around on both issues when I get a chance, DV.

**Comment by martanne** [comment]
Some updates from IRC investigations: the terminal has `VDSUSP` enabled, this can be checked by executing `stty -a` which will reveal something like `dsusp = ^Y`.

By default [libtermkey disables it](https://github.com/neovim/libtermkey/blob/205c743ad93bae3939e96c27ac2fbfc5a7008c73/termkey.c#L509-L511) (this also explains why the demo application works), but the relevant code path isn't executed within vis because we pass `TERMKEY_FLAG_NOTERMIOS` to `termkey_new`. This was added in fc0efcc89825ef1f61ffe5744b58f7d4e16c2aae to fix other job related issues.

To summarize: the problem is in the way terminal initialization is performed by the combination of libtermkey and curses. It does not occur on Linux because there `VDSUSP` is not supported.

**Comment by soveran** [comment]
A quick fix on the macos side is to run `stty dsusp undef` to unmap `^Y`.

**Comment by jocap** [comment]
I can confirm that this is a problem on OpenBSD as well. Perhaps there is some place in the documentation where this could be mentioned?

**Comment by malikbenkirane** [comment]
I can confirm the problem with OpenBSD, though @soveran workaround fits well to this issue although I still did not figured out why this setting is there by default.

I'd suggest - if actually feasible - as in `libtermkey` demo to unset that behaviour by default rather than adding documentation for the additional setup. I think it worths to provide compatible features across platforms "out of the box". Meanwhile this issue well suits for documentation.
